esp/src/src/react/recentFilters.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from "react";
+import Skeleton from '@material-ui/lab/Skeleton';
+import { Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Button, Typography, Paper } from "@material-ui/core";
+import { ThemeProvider } from "@material-ui/core/styles";
+import Tooltip from '@material-ui/core/Tooltip';
+import { theme } from './theme';
+import "dojo/i18n";
+// @ts-ignore
+import * as nlsHPCC from "dojo/i18n!hpcc/nls/hpcc";
+import { useGet } from "./hooks/useWsStore";
+
+interface RecentFilterProps {
+    ws_key: string;
+    widget: any;
```

Comment:
Instead of any it could be:  `{NewPage:any}`- not sure this will actually catch any mismatches though?

Reply:
Well since it is passing `this.widget` which gives me access to everything in that widget I felt that widget made more sense. Whereas NewPage is just describing the method I am looking for.